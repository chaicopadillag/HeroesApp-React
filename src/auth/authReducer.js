import { types } from '../types/types';

const authReducer = (state = {}, action) => {
	switch (action.type) {
		case types.login:
			return { ...action.payload, sesion: true };
		case types.logout:
			return { sesion: false };
		default:
			return state;
	}
};
export default authReducer;
