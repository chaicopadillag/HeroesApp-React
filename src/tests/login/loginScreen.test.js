import { mount } from 'enzyme';
import { AuthContext } from '../../auth/AuthContext';
import LoginScreen from '../../components/login/LoginScreen';

describe('Prueba en  <LoginScreen/>', () => {
	const history = {
		length: 10,
		push: jest.fn(),
		goBack: jest.fn(),
		replace: jest.fn(),
	};
	const contextvalue = {
		dispatch: jest.fn(),
		session: {
			sesion: false,
		},
	};
	const wrapper = mount(
		<AuthContext.Provider value={contextvalue}>
			<LoginScreen history={history} />
		</AuthContext.Provider>
	);

	test('debe de mostrar correctamente', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('debe de realizar el disptach y la navegacion', () => {
		wrapper.find('button').simulate('click');
		expect(contextvalue.dispatch).toHaveBeenCalledTimes(1);
		expect(history.replace).toHaveBeenCalledWith('/');
	});
});
