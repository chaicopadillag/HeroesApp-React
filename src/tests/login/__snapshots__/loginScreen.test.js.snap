// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en  <LoginScreen/> debe de mostrar correctamente 1`] = `
<div
  className="container mt-2"
>
  <h2>
    Login
  </h2>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login in App Heroes
  </button>
</div>
`;
